src/DynamoCoreWpf/ViewModels/Core/ConnectorViewModel.cs

```diff
@@ -1032,10 +1040,14 @@ private void PinViewModelPropertyChanged(object sender, PropertyChangedEventArgs
         {
             switch (e.PropertyName)
             {
-                case nameof(ConnectorPinModel.IsSelected):
+                case nameof(ConnectorPinViewModel.IsSelected):
                     var vm = sender as ConnectorPinViewModel;
                     AnyPinSelected = vm.IsSelected;
                     break;
+                case nameof(ConnectorPinViewModel.IsHoveredOver):
+                    vm = sender as ConnectorPinViewModel;
+                    AnyPinHoveredOver = vm.IsHoveredOver;
```

Comment:
Do you actually need to set this property like this?
Seems like you could just have

```csharp
internal bool IsAnyPinHovered => ConnectorPinViewCollection.Any(p => p.IsHoveredOver);
```

Or even just do the linq statement when you needed it, i cant really figure out how often this property is used

Reply:
Seems like the same could be done on the `AnyPinSelected`

Reply:
You would still need to wrap it in a handler right? Need to handle it every time those properties are updated.

Reply:
After looking at it seems like overkill to iterate thru all the pins in this scenario, I guess you only need the pin whose property changed.. no?

Reply:
Why do you need to handle it every time the property is updated? It looks to me like you only ever use this property in the 'CanInstantiateContextMenu' so don't see the need to constantly keeping track of it. Instead you can easily make that check only when you need it.

Reply:
^That goes for `AnyPinSelected`. Similar logic for `AnyPinHoveredOver`. However this flag tells us whether we can instantiate a ContextMenu for the wire. If `AnyPinHoveredOver` is true, then we cannot fire that command.

Reply:
Right, so what i am asking is, do you really need to constantly update this state, or could you not just on-demand check if any pins is hovered or selected, as in the Linq statement above?

I havent looked into the AnyPinSelected state so maybe thats different, but AnyPinHovered is used once in `CanInstantiateContextMenu` your current implementation does not require that state to be dynamically updated, you might as well call `ConnectorPinViewCollection.Any(p => p.IsHoveredOver);` there instead, that way you do not need to worry about these events.